source/renderer/app/components/staking/stake-pools/StakePools.js

```diff
@@ -160,50 +191,84 @@ export default class StakePools extends Component<Props, State> {
               search={search}
               onSearch={this.handleSearch}
               onClearSearch={this.handleClearSearch}
+              onGridView={this.handleGridView}
+              onListView={this.handleListView}
+              isListView={isListView}
+              isGridView={isGridView}
+              isFixed={(isFixed || maintainFixed) && !!filteredStakePoolsList.length}
               isClearTooltipOpeningDownward
+              isScrolled={isScrolled}
             />
-
-            {stakePoolsDelegatingList.length > 0 && (
+            {isListView && (
```

Comment:
@DeeJayElly Please, create dedicated components for list and grid view, so `StakePools.js` is more readable.

Reply:
`StakePoolsListView.js `

Reply:
I have already separated the logic on 2 components - StakePoolsTable.js & StakePoolsList.js